src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -487,19 +507,38 @@ export default class FilterOptionsComponent extends Component {
     return (option && filter) ? option.indexOf(filter) : -1;
   }
 
+  /**
+   * Clears all selected options. If this component is not part of a FilterBox,
+   * also send the REMOVED_FILTER analytics event. Otherwise, let FilterBox do it.
+   */
   clearOptions () {
+    if (!this.config.isDynamic) {
+      const removedFilterParam = this.config.optionType === 'RADIUS_FILTER'
+        ? this.getLocationRadiusFilterNode().getFilter().value
+        : JSON.stringify(this.getFilterNode().getFilter());
```

Comment:
Could we move the serialization of the removed filter into createRemovedFilterEvent as well? Then we wouldn't need to duplicate that code. I think it also fits within the responsibility of that method. It takes in SDK quantities (a Filter, fitlerType, etc. ) and serializes them so they can be understood by the Analytics backend.

Reply:
will do this when we eventually implement the analytics event